go/vt/vtgate/vindexes/vschema.go

```diff
@@ -331,9 +331,9 @@ func BuildVSchema(source *vschemapb.SrvVSchema, parser *sqlparser.Parser) (vsche
 		created:        time.Now(),
 	}
 	buildKeyspaces(source, vschema, parser)
-	// buildGlobalTables before buildReferences so that buildReferences can
+	// BuildGlobalTables before buildReferences so that buildReferences can
 	// resolve sources which reference global tables.
-	buildGlobalTables(source, vschema)
+	BuildGlobalTables(source, vschema)
 	buildReferences(source, vschema)
```

Comment:
in this sequence of operations, we remove auto increment table i.e. sequence tables from global tables on error.
The second call can add it back again.

look at `resolveAutoIncrement`

Reply:
But this problem exists with schema tracking too, right? It would add the table back to the vschema.

Reply:
In a way it is not a problem, as those tables added does not have the column vindex information, so no routing can happen on them.